docs/Guides/Write-Type-Provider.md

```diff
@@ -0,0 +1,32 @@
+<h1 align="center">Fastify</h1>
+
+## How to write your own type provider
+
+Things to keep in mind when implementing a custom [type provider](../Reference/Type-Providers.md):
+
+### Type Contravariance
+
+Whereas exhaustive type narrowing checks normally rely on `never` to represent
+an unreachable state, reduction in type provider interfaces should only be done
+up to `unknown`.
```

Comment:
Would it be more sensible to have this as another section in the Reference for type providers, rather than a guide? https://github.com/fastify/fastify/blob/9cd2fea549071f511a2a3b6b9304eb90d7cf46ec/docs/Reference/Type-Providers.md?plain=1#L3

Reply:
IMO it makes sense under the Type Providers section and add a "Customer Type Providers" or something similar..